pkg/scheduler/cache/status_updater/concurrency.go

```diff
@@ -120,12 +124,26 @@ func (su *defaultStatusUpdater) loadInflighUpdate(payload *updatePayload) (*infl
 	return data.(*inflightUpdate), true
 }
 
-func (su *defaultStatusUpdater) updatePod(
-	ctx context.Context, key updatePayloadKey, patchData []byte, subResources []string, object runtime.Object,
+func (su *defaultStatusUpdater) updatePodStatus(
+	ctx context.Context, key updatePayloadKey, object runtime.Object,
+) {
+	pod := object.(*v1.Pod)
+	_, err := su.kubeClient.CoreV1().Pods(pod.Namespace).UpdateStatus(
```

Comment:
I think it is less optimal to use this update function because it is very likely to fail if someone else updated the pod. I understand that the goal of this change is for this function to fail in a specific case, but it seems a waste of time to send it and wait for the failure if we can know that it will already fail.
My suggestion is that instead of this change, we will delete the updateData in the statusUpdater when calling to the Bound function.

Reply:
Maybe for the best results we should do both things ? both use `UpdateStatus` and also delete the updateData to prevent most of the failing attempts.